packages/cli/generators/relation/has-one-relation.generator.js

```diff
@@ -0,0 +1,200 @@
+// Copyright IBM Corp. 2020. All Rights Reserved.
+// Node module: @loopback/cli
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+'use strict';
+
+const path = require('path');
+const BaseRelationGenerator = require('./base-relation.generator');
+const relationUtils = require('./utils.generator');
+const utils = require('../../lib/utils');
+
+const CONTROLLER_TEMPLATE_PATH_HAS_ONE =
+  'controller-relation-template-has-one.ts.ejs';
+
+module.exports = class HasOneRelationGenerator extends BaseRelationGenerator {
+  constructor(args, opts) {
+    super(args, opts);
+  }
+
+  async generateControllers(options) {
+    this.artifactInfo.sourceModelClassName = options.sourceModel;
+    this.artifactInfo.targetModelClassName = options.destinationModel;
+    this.artifactInfo.sourceRepositoryClassName =
+      this.artifactInfo.sourceModelClassName + 'Repository';
+    this.artifactInfo.controllerClassName =
+      this.artifactInfo.sourceModelClassName +
+      this.artifactInfo.targetModelClassName +
+      'Controller';
+    this.artifactInfo.paramSourceRepository = utils.camelCase(
+      this.artifactInfo.sourceModelClassName + 'Repository',
+    );
+
+    this.artifactInfo.sourceModelName = utils.toFileName(options.sourceModel);
+    this.artifactInfo.sourceModelPath = utils.pluralize(
```

Comment:
@agnes512 I don't know where sourceModelPath and destinationModelPath are being used. Should I remove them?